File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.ts
function check(nums: number[]): boolean {
    let dip = -1;
    for (let i = 0; i < nums.length - 1; i++) {
        if (nums[i] > nums[i + 1]) {
            dip = i + 1;
            break;
        }
    }
    for (let i = 0; i < nums.length - 1; i++) {
        let index = ((i + dip) % nums.length);
        if (nums[index] > nums[(index + 1) % nums.length]) return false;
    }
    return true;
};

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int minIndex = -1;
        int min = Integer.MAX_VALUE;
        int n = nums.length;
        for(int i = 0; i < nums.length; i ++) {
            if((nums[i] < min) || (nums[i] == min && nums[i] != nums[i - 1])) {
                min = nums[i];
                minIndex = i;
            }
        }

        for(int i = 0; i < nums.length - 1; i ++) {
            int curIndex = i + minIndex;
            if(nums[curIndex % n] > nums[(curIndex + 1) % n]) return false;
        }
        return true;
    }
}